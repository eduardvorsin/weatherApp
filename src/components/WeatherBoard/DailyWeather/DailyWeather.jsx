import PropTypes from 'prop-types';
import { getFullDayDescription, getIconPathByCode, getWeatherStatusByCode } from '../../../utils/utils';
import classes from './DailyWeather.module.css';

export default function DailyWeather({
  temperatureMin,
  temperatureMax,
  day,
  weatherCode,
  className,
  testId,
}) {
  const iconPath = getIconPathByCode(weatherCode);
  const status = getWeatherStatusByCode(weatherCode);
  const date = getFullDayDescription(day);

  return (
    <div
      className={`${classes['daily-weather']} ${className}`}
      data-testid={testId}
    >
      <div
        className={classes['daily-weather__description']}
      >
        <p className={classes['daily-weather__text']}>
          {date}
        </p>
      </div>
      <img
        className={classes['daily-weather__icon']}
        src={iconPath}
        alt={status}
      />
      <p className={classes['daily-weather__temperature']}>
        <span
          className={classes['daily-weather__temperature--min']}
        >
          {temperatureMin}
          °
        </span>
        &nbsp;/&nbsp;
        <span
          className={classes['daily-weather__temperature--max']}
        >
          {temperatureMax}
          °
        </span>
      </p>
    </div>
  );
}

DailyWeather.defaultProps = {
  className: '',
  temperatureMin: 0,
  temperatureMax: 0,
  day: '2023-01-01T00:00',
  weatherCode: 0,
  testId: '',
};

DailyWeather.propTypes = {
  className: PropTypes.string,
  temperatureMin: PropTypes.number,
  temperatureMax: PropTypes.number,
  weatherCode: PropTypes.number,
  day: PropTypes.string,
  testId: PropTypes.string,
};
