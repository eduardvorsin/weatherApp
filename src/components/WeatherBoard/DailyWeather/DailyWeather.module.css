.daily-weather {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.daily-weather__description,
.daily-weather__temperature {
  flex: 0 1 calc(50% - 50px);
}

.daily-weather>*:not(:last-child) {
  margin-right: 15px;
}

.daily-weather__text {
  line-height: 1.3;
  font-size: 16px;
  margin: 0;
}

.daily-weather__icon {
  flex: 0 1 50px;
  height: 50px;
}

.daily-weather__temperature {
  margin: 0;
  text-align: right;
}

.daily-weather__temperature--max {
  color: var(--secondary-color)
}

@media(max-width:576px) {
  .daily-weather__icon {
    flex: 0 1 35px;
    height: 35px;
  }

  .daily-weather>*:not(:last-child) {
    margin-right: 10px;
  }
}

@media(max-width:769px) {

  .daily-weather__description,
  .daily-weather__temperature {
    flex: 0 1 130px;
  }
}
