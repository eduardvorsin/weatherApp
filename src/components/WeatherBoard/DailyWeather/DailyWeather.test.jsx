import { render, screen } from '@testing-library/react';
import DailyWeather from './DailyWeather';

describe('DailyWeather component tests', () => {
  it('renders correctly', () => {
    render(<DailyWeather
      className="test-class"
      temperatureMin={0}
      temperatureMax={21}
      day="2023-01-02T00:00"
    />);

    expect(screen.getByText(/21°/)).toBeInTheDocument();
    expect(screen.getByText(/0°/)).toBeInTheDocument();
    expect(screen.getByText(/понедельник, 2 января/)).toBeInTheDocument();
  });

  it('renders without props', () => {
    render(<DailyWeather />);

    expect(screen.getByText(/воскресенье, 1 января/)).toBeInTheDocument();
    expect(screen.getAllByText(/0°/)).toHaveLength(2);
  });

  it('the classname is correctly assigned to the element', () => {
    render(<DailyWeather
      className="test-class"
      testId="daily-weather"
    />);

    expect(screen.getByTestId('daily-weather')).toHaveClass('test-class');
  });

  it('snapshot with default data', () => {
    render(<DailyWeather
      testId="daily-weather"
    />);

    expect(screen.getByTestId('daily-weather')).toMatchSnapshot();
  });

  it('snapshot with mock text', () => {
    render(<DailyWeather
      className="test-class"
      temperatureMin={-23}
      temperatureMax={-6}
      day="2023-01-02T00:00"
      testId="daily-weather"
    />);

    expect(screen.getByTestId('daily-weather')).toMatchSnapshot();
  });
});
