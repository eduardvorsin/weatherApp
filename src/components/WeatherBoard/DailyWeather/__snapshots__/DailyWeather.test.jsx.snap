// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyWeather component tests snapshot with default data 1`] = `
<div
  class="daily-weather "
  data-testid="daily-weather"
>
  <div
    class="daily-weather__description"
  >
    <p
      class="daily-weather__text"
    >
      воскресенье, 1 января
    </p>
  </div>
  <img
    alt="ясно"
    class="daily-weather__icon"
    src="/assets/icons/sunny.svg"
  />
  <p
    class="daily-weather__temperature"
  >
    <span
      class="daily-weather__temperature--min"
    >
      0
      °
    </span>
     / 
    <span
      class="daily-weather__temperature--max"
    >
      0
      °
    </span>
  </p>
</div>
`;

exports[`DailyWeather component tests snapshot with mock text 1`] = `
<div
  class="daily-weather test-class"
  data-testid="daily-weather"
>
  <div
    class="daily-weather__description"
  >
    <p
      class="daily-weather__text"
    >
      понедельник, 2 января
    </p>
  </div>
  <img
    alt="ясно"
    class="daily-weather__icon"
    src="/assets/icons/sunny.svg"
  />
  <p
    class="daily-weather__temperature"
  >
    <span
      class="daily-weather__temperature--min"
    >
      -23
      °
    </span>
     / 
    <span
      class="daily-weather__temperature--max"
    >
      -6
      °
    </span>
  </p>
</div>
`;
